Cython/Utility/ModuleSetupCode.c

```diff
@@ -410,6 +410,16 @@
   #endif
 #endif
 
+#if defined (_MSC_VER)
```

Comment:
Is this the right place to define `__PYX_ENUM_DECL`?

Reply:
It might be better in https://github.com/cython/cython/blob/master/Cython/Utility/CppSupport.cpp (and then use that chunk of utility code only if needed)

Reply:
Thanks! Could you clarify what "only if needed" means here? Where would be best to use that utility code?

Reply:
Ideally you call `env.use_utility_code` at the point where you decide that you need it. However, you don't have a handle to a scope inside `declaration_code`. One option would be to set `utility_code` on the entry in `Symtab.Scope.declare_enum`?

If that doesn't work easily then it may be bad advice.

Reply:
Yes, attaching the utility code section to the C++ enum entry's `.utility_code` sounds good.

Reply:
Thanks both -- does what I have now seem better?

Reply:
What you've done was pretty much what I had in mind. Is `self.use_entry_utility_code(entry)` necessary? I'd have thought setting the attribute on the entry should be enough and using the utility code should automatically happen later (but I might be wrong)

Reply:
Hmm -- I tried and that doesn't work, unfortunately.